monitor/subscription.go

```diff
@@ -39,10 +39,28 @@ type NodeMonitor struct {
 	errHandlerCB     ErrHandler
 }
 
-// internal struct to manage various ids
-type itemIDs struct {
-	handle uint32 // client-provided
-	id     uint32 // from server
+// Item is a struct to manage Monitored Items
+type Item struct {
+	id     uint32     // from server
+	nodeID *ua.NodeID // from request
+	handle uint32     // client provided
+}
+
+// ID returns the MonitorItemID set by the server
+func (m Item) ID() uint32 {
```

Comment:
should `m` be a pointer? Or is there a reason these are value receivers?

Reply:
Fixed this now. I see that you use pointers all over and changed this PR to follow the same style. Just out of curiosity, Is there a good reason for the extended pointer use?

Reply:
This is a pretty good summary: https://github.com/golang/go/wiki/CodeReviewComments#receiver-type

My approach tends to be: pointer unless I have an explicit reason to deviate (which generally only shows up when trying optimize the performance of a function),